ibis/backends/tests/test_vectorized_udf.py

```diff
@@ -73,26 +137,65 @@ def overwrite_struct_analytic(v, w):
     return v - v.mean(), w - w.mean()
 
 
-@analytic(
-    input_type=[dt.double, dt.double],
-    output_type=dt.Struct(['demean', 'demean_weight'], [dt.double, dt.double]),
-)
+# analytic multi-column UDF
 def demean_struct(v, w):
     assert isinstance(v, pd.Series)
     assert isinstance(w, pd.Series)
     return v - v.mean(), w - w.mean()
 
 
-@reduction(
-    input_type=[dt.double, dt.double],
-    output_type=dt.Struct(['mean', 'mean_weight'], [dt.double, dt.double]),
-)
+def create_demean_struct_udf(result_formatter):
+    return analytic(
+        input_type=[dt.double, dt.double],
+        output_type=dt.Struct(
+            ['demean', 'demean_weight'], [dt.double, dt.double]
+        ),
+    )(_format_struct_udf_return_type(demean_struct, result_formatter))
+
+
+demean_struct_udfs = [
+    create_demean_struct_udf(
+        result_formatter=lambda v1, v2: (v1, v2)
+    ),  # tuple of pd.Series,
+    create_demean_struct_udf(
+        result_formatter=lambda v1, v2: [v1, v2]
+    ),  # list of pd.Series,
+    create_demean_struct_udf(
+        result_formatter=lambda v1, v2: (np.array(v1), np.array(v2))
+    ),  # tuple of np.array,
+    create_demean_struct_udf(
+        result_formatter=lambda v1, v2: [np.array(v1), np.array(v2)]
+    ),  # list of np.array,
+]
+
+
+# reduction multi-column UDF
 def mean_struct(v, w):
     assert isinstance(v, (np.ndarray, pd.Series))
     assert isinstance(w, (np.ndarray, pd.Series))
     return v.mean(), w.mean()
 
 
+def create_mean_struct_udf(result_formatter):
+    return reduction(
+        input_type=[dt.double, dt.double],
+        output_type=dt.Struct(['mean', 'mean_weight'], [dt.double, dt.double]),
+    )(_format_struct_udf_return_type(mean_struct, result_formatter))
+
+
+mean_struct_udfs = [
```

Comment:
Series can also be supported here I think:

pd.Series([v1, v2])

Reply:
Updated to include a test for this as well.

Reply:
Hmm maybe this isn't directly supported in the dask backend, will have to take a look.